appinventor/blocklyeditor/src/generators/yail/helpers.js

```diff
@@ -44,3 +44,22 @@ Blockly.Yail['helpers_assets'] = function() {
   }
   return [Blockly.Yail.quote_(field.getValue()), Blockly.Yail.ORDER_ATOMIC];
 }
+
+Blockly.Yail['helpers_providermodel'] = function() {
+  var field = this.getField('PROVIDERMODEL');
+  var fielddisplayvalue = field.getValue();
+  var fieldvalue = top.chatproxyinfo['model'][fielddisplayvalue];
+  if (!field) {
```

Comment:
It seems like this check should come before `field` gets used on the previous two lines.